server/src/main/java/org/opensearch/search/backpressure/settings/SearchBackpressureSettings.java

```diff
@@ -46,6 +46,15 @@ private static class Defaults {
     public static final Setting<String> SETTING_MODE = Setting.simpleString(
```

Comment:
Can we replace this with a typed setting per the [recommendation here](https://github.com/opensearch-project/OpenSearch/blob/c765e3a4bfb4e2d2532e1afde34758e48c644b3d/server/src/main/java/org/opensearch/common/settings/Setting.java#L94-L102):

```
 * It's recommended to use typed settings rather than string based settings. For example adding a setting for an enum type:
 * <pre>{@code
 * public enum Color {
 *     RED, GREEN, BLUE;
 * }
 * public static final Setting<Color> MY_BOOLEAN =
 *     new Setting<>("my.color.setting", Color.RED.toString(), Color::valueOf, Setting.Property.NodeScope);
 * }
 * </pre>
```

Reply:
Done!